ros/processor/inventory_events_consumer.py

```diff
@@ -95,9 +96,11 @@ def host_delete_event(self, msg):
             LOG.debug(
                 f"{self.prefix} - Received a message for system with inventory_id {host_id}"
             )
-            rows_deleted = db.session.query(System.id).filter(System.inventory_id == host_id).delete()
+
+            rows_deleted = db.session.execute(db.delete(System).filter(System.inventory_id == host_id))
             db.session.commit()
-            if rows_deleted == 1:
+
+            if rows_deleted.rowcount > 0:
```

Comment:
Is `rowcount` pre-defined anywhere?

And the second thing is, I see you have changed the condition from `==1` to `>0`. Any reason or difference in the result due to this check?

Reply:
The rowcount is attribute on sqlalchemy object,

~~~
(Pdb) rows = db.session.execute(db.delete(System).filter(System.id == 1))
(Pdb) rows.rowcount
1
(Pdb) rows.rowcount
1
(Pdb) rows.rowcount
1
(Pdb) rows.__class__
<class 'sqlalchemy.engine.cursor.LegacyCursorResult'>
(Pdb) dir(rows)
['returns_rows', 'rowcount', 'scalar', 'scalar_one', 'scalar_one_or_none', 'scalars', 'supports_sane_multi_rowcount', 'supports_sane_rowcount', 'unique', 'yield_per']
(Pdb) rows = db.session.execute(db.delete(System).filter(System.id == 1))
(Pdb) rows.rowcount
0
~~~

and the below query can only delete one row considering inventory_id is unique and query has the '==' constraint?

~~~
            rows_deleted = db.session.execute(db.delete(System).filter(System.inventory_id == host_id))
~~~

So I think >0 is sane as well? however I can restore it back to '==0' if you think we should be very explicit about it.

Reply:
I think `==1` makes sense. Could you revert it back?